lxd/instance/drivers/driver_common.go

```diff
@@ -1017,3 +1017,63 @@ func (d *common) resetInstanceID() error {
 
 	return nil
 }
+
+// needsNewInstanceID checks the changed data in an Update call to determine if a new instance-id is necessary.
+func (d *common) needsNewInstanceID(changedConfig []string, oldExpandedDevices deviceConfig.Devices) bool {
+	// Look for cloud-init related config changes.
+	for _, key := range []string{
+		"cloud-init.vendor-data",
+		"cloud-init.user-data",
+		"cloud-init.network-config",
+		"user.vendor-data",
+		"user.user-data",
+		"user.network-config",
+	} {
+		if shared.StringInSlice(key, changedConfig) {
+			return true
+		}
+	}
+
+	// Look for changes in network interface names.
+	getNICNames := func(devs deviceConfig.Devices) []string {
+		names := []string{}
```

Comment:
Could replace this with `names := make([]string, 0, len(des))` to be more efficient when allocating the slice.

Reply:
No, that would cause a slice that's too large in most cases.